Source/UZKArchive.m

```diff
@@ -571,15 +567,12 @@ - (BOOL)extractFilesTo:(NSString *)destinationDirectory
                                        detail:detail];
                             return;
                         }
-                    } else if (!isDirectory) {
-                        NSString *detail = [NSString localizedStringWithFormat:NSLocalizedStringFromTableInBundle(@"Extract path exists, but is not a directory: %@", @"UnzipKit", _resources, @"Detailed error string"),
-                                            extractDir];
-                        UZKLogError("UZKErrorCodeOutputErrorPathIsAFile: %{public}@", detail);
-                        [welf assignError:&strongError code:UZKErrorCodeOutputErrorPathIsAFile
-                                   detail:detail];
-                        return;
                     }
-
+                    
+                    if (info.isDirectory) {
+                        UZKLogDebug("Created empty directory")
```

Comment:
Why not creating empty directory here?

Reply:
It was already created on line 558. By the time we get here, there's nothing left to do, so we continue.

Reply:
That one is for extraction root directory. This one would be for empty directories inside zip file.

Reply:
No, it's not just the extraction directory - look where `extractDir` is defined - if the current item is a directory, `extractDir` is its path, and if it's a file, `extractDir` is its parent directory. It's creating all intermediate directories.

Reply:
Thanks